buildSrc/subprojects/plugins/plugins.gradle.kts

```diff
@@ -12,6 +12,7 @@ dependencies {
     implementation("com.google.code.gson:gson:2.7")
     testImplementation("junit:junit:4.12")
     testImplementation("com.nhaarman:mockito-kotlin:1.6.0")
+    testImplementation("io.mockk:mockk:1.8.13")
```

Comment:
❌ Looks like now we have two mocking libraries: `mockito-kotlin` and `mockk`. I think we should choose one. What is the difference between them?

Reply:
IIUC, `mockito-kotlin` is just a "Kotlin DSL" of mockito. It has no idea of Java/Kotlin and can only mock non-final `protected`/`public` methods.

`mockk` is a little more powerful kotlin-dedicated mocking library which can mock extension functions - this is what we need in our test. I think we can unify to `mockk`.

Reply:
I don't see any problems if we change all mock to `mockk`. @bamboo WDYT?

Reply:
`mockk` seems to have all the features we currently rely on in the `kotlin-dsl` tests.

I have no strong feelings one way or the other.

Reply:
Well, no strong feelings but I should say I'm bit afraid of things like `mockkStatic`.

Reply:
@bamboo do you have better ideas in this case? Can we mock extension methods in other way, or change the schema to avoid such mocking?

Reply:
From my personal experience it seems like doing something wrong when mocking static methods. Extension methods are basically static methods, so I am not sure we should try to mock them and introduce a separate service which then can be mocked instead.

Apart from that, I had also quite some hard to diagnose test failures when using PowerMock for mocking static methods and the test become slower.

Reply:
I'm not sure if there's any hack in classloader like powermock does. Let me take a quick look at `mockk` implementation.

Reply:
I had a vague memory of PowerMock issues years ago but forgot the details. PowerMock is doing the mock by a hack classloader which modifies the bytecode it loads. According to [mockk's code](https://github.com/mockk/mockk/blob/6005111439fde78521aa4dce0b53ec63cd882853/agent/jvm/src/main/kotlin/io/mockk/proxy/jvm/transformation/InlineInstrumentation.kt#L39), mockk is doing this work by [JVM Instrumentation](https://docs.oracle.com/javase/8/docs/api/java/lang/instrument/Instrumentation.html#retransformClasses-java.lang.Class...-) - which seems to be fine.

Back to the question whether we should mock extension/static methods, I was hesitating a lot. We're using Kotlin extension methods really a looooot and I believe this will be an issue sooner or later - I don't want to change production code just to let test code happy. 

For now, my opinion is, if this `mockkStatic` has no impacts on the code outside the test class (that's what I understand, is this true?), let's use it, unless we find any evidence that it's not suitable. 

@bamboo @eskatos Would like to hear you experts' opinions.